### ISSUE TITLE
Headless won't start on ARM

### ISSUE THREAD

**Issue by deltabotx** [Yellow-Dog-Man/Resonite-Issues#3115]
### Describe the bug?

Attempting to start the latest headless client version(2024.10.23.15) on ARM yields the error message "Unhandled exception. System.IO.FileLoadException: Could not load file or assembly 'Resonite, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'.
Aborted". Headless worked fine before this update.

### To Reproduce

Load headless files onto ARM Linux client (Ubuntu Server on Oracle Cloud Hosting's Ampere A1 client in my case), Custom compile all required dependencies for system hardware, run headless using command "Sudo dotnet Resonite.dll"

### Expected behavior

Headless client starts and runs as it did prior to update.

### Screenshots

![Image](https://github.com/user-attachments/assets/48d2e9af-c020-43ec-99aa-bbf52af8db87)

### Resonite Version Number

2024.10.23.15

### What Platforms does this occur on?

Linux

### What headset if any do you use?

Valve Index

### Log Files

Client will not start.

### Reporters

discord users: @deltabotx, @gamethecupdog, @tiky, @adamski2003

**Comment by ticpu** [comment]
From what I can gather, it seems the Resonite.dll file has changed from being a neutral .Net assembly to being a x86-64 only .Net assembly.  It is now listed alongside Steamworks.NET.dll in the list of x86-64 only libraries by the file utility.
![Image](https://github.com/user-attachments/assets/90655335-1ac4-4f7b-a41b-4dabd76fa4a5)

**Comment by Frooxius** [comment]
We do not officially support headless on ARM right now, so this might not be a priority if it's too complicated to fix.

This might be something to do potentially with the new SDK build system. I'm looping in @BlueCyro to this since he handled that, maybe he has some ideas there.

**Comment by BlueCyro** [comment]
MSBuild handles neutral runtimes and dependencies differently than the SDK-style projects do.

When building newer .NET projects on the SDK format (such as the headless, which is now .NET 8), it will include dlls for all of the possible runtimes so that the program can work on a multitude of different platforms. However, it completely bloats the deps.json with inherited dependencies and causes the headless to search in the wrong directories for the native libraries.

Due to some leftover baggage from the old build system, those native DLLs are provided manually. The internal FrooxEngine project is still quite vast and it'll take some time before I can figure out why building a platform-neutral build of the headless causes so much trouble. Building it with specified runtime identifier was the cleanest way to retain the officially supported functionality for the time-being.

I don't think this'll be a quick fix. The build system is still in-need of some polish after basically re-doing the whole thing.

**Comment by Redd56** [comment]
OH this one can probably be closed since we now got headless on arm right @jae1911?

**Comment by jae1911** [comment]
Correct, ARM support landed last December and somehow this issue was missed, closing.
